### DISCUSSION TITLE
Live market data with custom fields

### DISCUSSION THREAD

**Discussion by npajkovsky** [wboayue/rust-ibapi#178] [Q&A] [answered]
Hello,

does TWS API support similar field selectors like this Web API?

https://www.interactivebrokers.com/campus/ibkr-api-page/webapi-ref/#tag/Trading-Market-Data/paths/~1iserver~1marketdata~1snapshot/get

**Comment by wboayue** [comment, marked as answer]
It looks like the Web API does some extra things like allowing you to specify multiple contract IDs. The API to use for TWS is the [market data API](https://docs.rs/ibapi/latest/ibapi/client/struct.Client.html#method.market_data). You can request different tick types using the `generic_ticks` parameter. Tick types are documented [here](https://www.interactivebrokers.com/campus/ibkr-api-page/twsapi-doc/#available-tick-types). Specifying generic ticks is not as fine grained as the field level filtering on the Web API.
